app/controllers/review_mapping_controller.rb

```diff
@@ -212,82 +245,127 @@ def add_metareviewer
     redirect_to action: 'list_mappings', id: mapping.assignment.id, msg: msg
   end
 
+  # This method is used to assign a metareviewer dynamically. 
+  # The method first finds the assignment using the assignment_id param. 
+  # It then finds the metareviewer from AssignmentParticipant. The method then
+  # assigns the metareviewer to the asignment dynamically. If the operation fails
+  # an error is thrown. 
   def assign_metareviewer_dynamically
     assignment = Assignment.find(params[:assignment_id])
     metareviewer = AssignmentParticipant.where(user_id: params[:metareviewer_id], parent_id: assignment.id).first
-
-    assignment.assign_metareviewer_dynamically(metareviewer)
-
+    begin
+      assignment.assign_metareviewer_dynamically(metareviewer)
+    rescue StandardError => e
+      flash[:error] = e
+    end
     redirect_to controller: 'student_review', action: 'list', id: metareviewer.id
   end
 
+  # This method gets the reviewer form the AssignmentParticipant. 
+  # If this operation fails, the code handles it by throwing an error. 
+  # If the reviewer does not exist, an error is thrown, asking the user to
+  # register the user first. 
+  # @param user is {Participant} type.
+  # @param reg_url is {string} 
   def get_reviewer(user, assignment, reg_url)
-    reviewer = AssignmentParticipant.where(user_id: user.id, parent_id: assignment.id).first
-    raise "\"#{user.name}\" is not a participant in the assignment. Please <a href='#{reg_url}'>register</a> this user to continue." if reviewer.nil?
-
-    reviewer.get_reviewer
-  rescue StandardError => e
-    flash[:error] = e.message
+    begin
+      reviewer = AssignmentParticipant.where(user_id: user.id, parent_id: assignment.id).first
```

Comment:
Placing AssignmentParticipant object in `reviewer` is misleading. Can you rename this variable name?